pxr/usd/usdGeom/testenv/testUsdGeomTriangulation.py

```diff
@@ -0,0 +1,64 @@
+#!/pxrpythonsubst
+#
+# pylint: disable=range-builtin-not-iterating
+
+from __future__ import print_function
+from pxr import Usd, UsdGeom, Vt
+import unittest
+
+
+class testUsdGeomTriangulation(unittest.TestCase):
+
+    def test_Concave(self):
+        stage = Usd.Stage.Open('Concave.usda')
+        prim = stage.GetPrimAtPath("/root/Concave")
+        mesh = UsdGeom.Mesh(prim)
+        result = mesh.FanTriangulate()
+        self.assertTrue(result)
+        self.assertEqual(mesh.GetFaceVertexCountsAttr().Get(), Vt.IntArray([4]))
+        self.assertEqual(mesh.GetFaceVertexIndicesAttr().Get(), Vt.IntArray([2, 3, 1, 0]))
+
+    def test_Convex(self):
+        stage = Usd.Stage.Open('Convex.usda')
+        prim = stage.GetPrimAtPath("/root/Convex")
+        mesh = UsdGeom.Mesh(prim)
+        result = mesh.FanTriangulate()
+        self.assertTrue(result)
+        self.assertEqual(mesh.GetFaceVertexCountsAttr().Get(), Vt.IntArray([4]))
+        self.assertEqual(mesh.GetFaceVertexIndicesAttr().Get(), Vt.IntArray([0, 2, 3, 1]))
+
+    def test_Faces3D(self):
+        stage = Usd.Stage.Open('Faces3D.usda')
+        prim = stage.GetPrimAtPath("/root/Faces3D")
+        mesh = UsdGeom.Mesh(prim)
+        result = mesh.FanTriangulate()
+        self.assertTrue(result)
+        self.assertEqual(mesh.GetFaceVertexCountsAttr().Get(), Vt.IntArray(
```

Comment:
since FanTriangulate yields a stable ordering, a nicety IMO would be to save the triangulation in Faces3D.usda, perhaps at /root/Faces3DTriangulated, or some such. If it was spatially offset from the Faces3D prim we'd get a nice visual a/b in usdview in addition to supporting the unit tests.

Reply:
Changed since last version, the tests compares against images.